sockeye/layers.py

```diff
@@ -123,46 +123,34 @@ class OutputLayer:
     """
     Defines the output layer of Sockeye decoders. Supports weight tying and weight normalization.
 
-    :param num_hidden: Number of hidden units in layer input (decoder representation).
-    :param num_embed: Target embedding size.
+    :param hidden_size: Decoder hidden size.
     :param vocab_size: Target vocabulary size.
-    :param weight_tying: Whether to use target embedding parameters as output layer parameters.
-    :param embed_weight: Optional embedding matrix. Required if weight_tying == True.
     :param weight_normalization: Whether to apply weight normalization.
     :param prefix: Prefix used for naming.
     """
 
     def __init__(self,
-                 num_hidden: int,
-                 num_embed: int,
+                 hidden_size: int,
                  vocab_size: int,
-                 weight_tying: bool,
-                 embed_weight: Optional[mx.sym.Symbol],
-                 weight_normalization: bool,
-                 prefix: str = '') -> None:
+                 weight: Optional[mx.sym.Symbol],
+                 weight_normalization: bool) -> None:
         self.vocab_size = vocab_size
 
-        if weight_tying:
-            utils.check_condition(num_hidden == num_embed,
-                                  "Weight tying requires target embedding size and decoder hidden size " +
-                                  "to be equal: %d vs. %d" % (num_embed, num_hidden))
-
-            logger.info("Tying the target embeddings and prediction matrix.")
-            assert embed_weight is not None, "Must provide embed_weight if weight_tying == True"
-            self.w = embed_weight
+        if weight is None:
+            self.w = mx.sym.Variable("target_output_weight", shape=(vocab_size, hidden_size))
```

Comment:
why are you no longer adding the prefix? Who is to say that you wouldn't maybe want multiple output layers in the future.

Reply:
Will add a prefix argument to the OutputLayer constructor